File: python/custom-set/custom_set.py
class CustomSet:
    def __init__(self, elements=None):
        if not elements: elements = []
        self.set = set()
        for e in elements:
            self.add(e)

    def __len__(self):
        return len(self.set)

    def isempty(self):
        return len(self.set) == 0

    def __contains__(self, element):
        return element in self.set

    def issubset(self, other):
        for o in self.set:
            if o not in other.set: return False
        return True

    def isdisjoint(self, other):
        if not other: return True
        for o in other.set:
            if o in self.set: return False
        return True

    def __eq__(self, other):
        if len(other) != len(self): return False
        for o in other.set:
            if o not in self.set: return False
        return True

    def add(self, element):
        if element not in self.set: self.set.add(element)

    def intersection(self, other):
        new_set = CustomSet()
        for e in self.set:
            if e in other.set: new_set.add(e)
        return new_set

    def __sub__(self, other):
        new_set = CustomSet()
        new_set += self
        for o in other.set:
            if o in new_set.set: new_set.set.remove(o)
        return new_set

    def __add__(self, other):
        new_set = CustomSet()
        for o in other.set: new_set.add(o)
        for o in self.set: new_set.add(o)
        return new_set

File: swift/custom-set/Sources/CustomSet/CustomSet.swift
class CustomSet: Equatable {
    var set:Set<Int>
    var isEmpty:Bool { self.set.count == 0 }

    init(_ set:[Int]) { self.set = Set(set) }

    func add(_ val:Int) { set.insert(val) }

    func contains(_ val:Int)        -> Bool { self.set.contains(val) }
    func isSubset(of:CustomSet)     -> Bool { set.isSubset(of:of.set) }
    func isDisjoint(with:CustomSet) -> Bool { with.set.intersection(set).count == 0 }
    static func == (lhs:CustomSet, rhs:CustomSet) -> Bool { lhs.set == rhs.set }

    func intersection(_ with:CustomSet) -> CustomSet { CustomSet(Array(set.intersection(with.set))) }
    func difference(_ of:CustomSet)     -> CustomSet { CustomSet(Array(set.subtracting(of.set))) }
    func union(_ with:CustomSet)        -> CustomSet { CustomSet(Array(set.union(with.set))) }
}
